src/storage/btree.c

```diff
@@ -14399,9 +14420,31 @@ btree_find_boundary_leaf (THREAD_ENTRY * thread_p, BTID * btid, VPID * pg_vpid,
 
       btree_read_fixed_portion_of_non_leaf_record (&rec, &nleaf);
       C_vpid = nleaf.pnt;
-      C_page = pgbuf_fix (thread_p, &C_vpid, OLD_PAGE, PGBUF_LATCH_READ, PGBUF_UNCONDITIONAL_LATCH);
+      C_page = pgbuf_fix_read_old_and_check_repl_desync (thread_p, C_vpid, PGBUF_UNCONDITIONAL_LATCH);
```

Comment:
This can also be in another function, but this one can only wait for LSA replication, without the retry.
The retry is only a flag which says the caller that it should re-try from start.

I guess a goto may be unavoidable :(